modules/lkqdBidAdapter.js

```diff
@@ -0,0 +1,208 @@
+      bidRequest.sizes = [[adWidth, adHeight]];
+    }
+
+    for (var j = 0; j < bidRequest.sizes.length; j++) {
+      var size = bidRequest.sizes[j];
+      var playerWidth;
+      var playerHeight;
+      if (size && size.length == 2) {
+        playerWidth = size[0];
+        playerHeight = size[1];
+      } else {
+        utils.logWarn('Warning: Could not determine width/height from the provided adUnit');
+      }
+
+      var sspUrl = ENDPOINT.concat();
+
+      // required parameters
+      sspUrl = _replaceMacro(PID_KEY, bidRequest.params.placementId, PID_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(SID_KEY, bidRequest.params.siteId, SID_DEFAULT, sspUrl);
+      // optional parameters
+      sspUrl = _replaceMacro(OUTPUT_KEY, bidRequest.params.output, OUTPUT_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(EXECUTION_KEY, bidRequest.params.execution, EXECUTION_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(PLACEMENT_KEY, bidRequest.params.placement, PLACEMENT_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(PLAYINIT_KEY, bidRequest.params.playinit, PLAYINIT_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(VOLUME_KEY, bidRequest.params.volume, VOLUME_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(TIMEOUT_KEY, bidRequest.params.timeout, TIMEOUT_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(WIDTH_KEY, playerWidth, WIDTH_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(HEIGHT_KEY, playerHeight, HEIGHT_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(DNT_KEY, bidRequest.params.dnt, DNT_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(PAGEURL_KEY, bidRequest.params.pageurl, PAGEURL_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(CONTENTID_KEY, bidRequest.params.contentId, CONTENTID_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(CONTENTTITLE_KEY, bidRequest.params.contentTitle, CONTENTTITLE_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(CONTENTLENGTH_KEY, bidRequest.params.contentLength, CONTENTLENGTH_DEFAULT, sspUrl);
+      sspUrl = _replaceMacro(CONTENTURL_KEY, bidRequest.params.contentUrl, CONTENTURL_DEFAULT, sspUrl);
+      // random number to prevent caching
+      sspUrl = sspUrl + '‌&rnd=' + Math.floor(Math.random() * 999999999);
+
+      var sspData = {};
+      sspData.bidId = bidRequest.bidId;
+      sspData.bidWidth = playerWidth;
+      sspData.bidHeight = playerHeight;
+
+      bidRequests.push({
+        method: 'GET',
+        url: sspUrl,
+        data: sspData
+      });
+    }
+  }
+
+  return bidRequests;
+}
+
+function interpretResponse(serverResponse, bidRequest) {
+  var bidResponses = [];
+  if (serverResponse && serverResponse.body) {
+    if (serverResponse.error) {
+      utils.logError('Error: ' + serverResponse.error);
+      return bidResponses;
+    } else {
+      try {
+        var bidResponse = {};
+        if (bidRequest && bidRequest.data && bidRequest.data.bidId && bidRequest.data.bidId !== '') {
+          var sspXml = new window.DOMParser().parseFromString(serverResponse.body, 'text/xml');
```

Comment:
What happens if this returns error. I think line 175, 179, 180 will throw error. 
https://developer.mozilla.org/en-US/docs/Web/API/DOMParser

Also Can you also confirm your unit tests are passing in Mobile Safari. I am not sure about its support